### ISSUE TITLE
Fatal: repository contains errors

### ISSUE THREAD

**Issue by domsl** [restic/restic#2889]
Output of `restic version`
--------------------------

restic 0.8.1 (v0.8.1-121-g94ec55f)
compiled with go1.8.3 on linux/amd64

How did you run restic exactly?
-------------------------------
```

/opt/restic/restic -r ~/Backups/bckp_files/ -p ~/Backups/pw.txt forget --keep-hourly 25 --keep-daily 8 --keep-weekly 5 --keep-monthly 13 --keep-yearly 10 --prune

/opt/restic/restic -r ~/Backups/bckp_files/ -p ~/Backups/pw.txt check
```

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
This is running locally on a ubuntu server. The data gets mostly transferred in by syncthing.

Expected behavior
-----------------
No errors while running prune and check.

Actual behavior
---------------

```
……
……
148 snapshots have been removed, running prune
counting files in repo
building new index for repo
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 487.830364ms: offset is negative
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 822.224653ms: offset is negative
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 623.82462ms: offset is negative
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 845.7124ms: offset is negative
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 3.474935465s: offset is negative
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 2.929109904s: offset is negative
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 5.923026415s: offset is negative
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 7.98542713s: offset is negative
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 15.254662208s: offset is negative
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 10.721439453s: offset is negative
pack file cannot be listed 88f6559fef38c0eaaf8c2dba38bfa73d9ef3df932f3c61bc1832fee76aa911ea: offset is negative
[3:59] 100.00%  43565 / 43565 packs
repository contains 43564 packs (319287 blobs) with 197.607 GiB
processed 319287 blobs: 0 duplicate blobs, 0B duplicate
load all snapshots
find data that is still in use for 342 snapshots
[0:32] 100.00%  342 / 342 snapshots
found 318012 of 319287 data blobs still in use, removing 1275 blobs
will remove 0 invalid files
will delete 49 packs and rewrite 473 packs, this frees 556.196 MiB
[1:40] 100.00%  473 / 473 packs rewritten
counting files in repo
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 688.899355ms: offset is negative
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 788.071133ms: offset is negative
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 565.48697ms: offset is negative
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 1.054524281s: offset is negative
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 2.715452968s: offset is negative
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 4.872357729s: offset is negative
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 6.444036194s: offset is negative
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 12.422740982s: offset is negative
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 18.725286606s: offset is negative
Load(<data/88f6559fef>, 591, -3) returned error, retrying after 20.386167523s: offset is negative
pack file cannot be listed 88f6559fef38c0eaaf8c2dba38bfa73d9ef3df932f3c61bc1832fee76aa911ea: offset is negative
[3:44] 100.00%  43347 / 43347 packs
finding old index files
saved new indexes as [ce3c72e8 881031ff 3718f09e 432dcf79 afb1e293 1f6dcceb fb11b318 2b8e4e74 c70174b0 865b582a d368bd16 08a186c9 8bc6734a 36bd6737 2771b856]
remove 56 old index files
[0:01] 100.00%  522 / 522 packs deleted
done
password is correct
create exclusive lock for repository
load indexes
check all packs
pack 88f6559fef38c0eaaf8c2dba38bfa73d9ef3df932f3c61bc1832fee76aa911ea: not referenced in any index
check snapshots, trees and blobs

Fatal: repository contains errors
```

Steps to reproduce the behavior
-------------------------------

re run the commands

Do you have any idea what may have caused this?
-----------------------------------------------

No, unfortunately.

Do you have an idea how to solve the issue?
-------------------------------------------

Also no.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

Not today, but I already used it once to recover some files and I sleep much better knowing there are incremental backups of my files synced in via syncthing to a server.
Restic is great. :)

I am not sure if I should report this as a bug. But nevertheless, I got this scary looking error and would really appreciate away to fix my backup.

**Comment by rawtaz** [comment]
@domsl Restic version 0.8.1 is from december 2017, you should retry this with a newer restic version. You could either use the latest release (0.9.6) but I'd recommend that you instead use the latest master build from [here](https://beta.restic.net/?sort=time&order=desc). Please update the post with the new information and output. Let us know how it goes!

**Comment by domsl** [comment]
Thanks - will do and report back! :)

**Comment by MichaelEischer** [comment]
@domsl The offset error is identical to #1633 which was fixed by #1635 and is included in restic >= v0.8.3.
Your restic version `v0.8.1-121-g94ec55f` already contains 953f3d55ee62bee42d4fdb8b774d980255bfd29b which introduced the offset bug.

**Comment by domsl** [comment]
```
repository contains 43387 packs (318367 blobs) with 197.134 GiB
processed 318367 blobs: 0 duplicate blobs, 0 B duplicate
load all snapshots
find data that is still in use for 336 snapshots
[0:25] 100.00%  336 / 336 snapshots
found 317693 of 318367 data blobs still in use, removing 674 blobs
will remove 0 invalid files
will delete 18 packs and rewrite 274 packs, this frees 317.066 MiB
[1:20] 100.00%  274 / 274 packs rewritten
counting files in repo
[3:35] 100.00%  43290 / 43290 packs
finding old index files
saved new indexes as [a1616073 f85dac76 038a8ceb 808f16a5 01f8f9ce f0c280fa a8bf5bf2 d93d1147 9b769e09 44e16819 5f3d3912 f5c7a8d1 0d658155 c4e36227 7cc6a65f]
remove 28 old index files
[0:00] 100.00%  28 / 28 files deleted
remove 292 old packs
[0:00] 100.00%  292 / 292 files deleted
done
using temporary cache in /tmp/restic-check-cache-421118841
repository ff692fb1 opened successfully, password is correct
created new cache in /tmp/restic-check-cache-421118841
create exclusive lock for repository
load indexes
check all packs
check snapshots, trees and blobs
no errors were found
```

Happy to report no errors found after the upgrade to the new version. :)
